### DISCUSSION TITLE
[made with Redbean] DirectoryTG.com

### DISCUSSION THREAD

**Discussion by secjoa** [jart/cosmopolitan#1245] [Show and tell]
Hello!

I created my website using [Redbean](https://redbean.dev/) + [fullmoon](https://github.com/pkulchenko/fullmoon) + [postgres](https://github.com/HereAdvertise/pgmoon).

For those who want to visit: [www.directorytg.com](http://www.directorytg.com/) (a directory of Telegram channels and groups)

Redbean is a great framework with low memory and CPU usage.

Thanks!

**Comment by jart** [comment]
That's awesome, thanks for sharing. I see you're using a Cloudflare proxy. Have you considered exposing it to the web directly? I actually switched some of our services *away* from Cloudflare because Redbean's token bucket / blackholed solution works better. It does however assume you're on a mighty network like Google's and that ingress is free. If you're self-hosting then you're probably better off with cloudflare unfortunately.

**Comment by secjoa** [comment]
I considered not using Cloudflare. However, I chose to use it because I utilize other resources like web analytics, certificates, registrar, and Turnstile. I host my site on a Digital Ocean droplet.
Regarding hosting, I use a very cheap droplet ($4/month). Observing through htop, the redbean consumption is almost zero.
My intention is for my site to grow gradually. I believe I won't have issues scaling the server. I came from using OpenResty + Lapis. Even though OpenResty is fast, I had problems scaling it. Switching to Redbean was the best choice. Another important point was the learning curve; I practically had no problems. I also really liked the fork() scheme.
